Source/EasyNetQ/Consumer/InternalConsumer.cs

```diff
@@ -133,11 +142,16 @@ public void HandleBasicCancelOk(string consumerTag)
             Cancel();
         }
 
+        /// <summary>
+        /// If we get this from the server, it means, that something went wrong as per the protocol documentation,
```

Comment:
I'm not sure about this, I've done some testing and we're completely changing how it behaves the consumer cancellation, `HandleBasicCancel` is not sent only when something goes wrong, for instance try to delete a queue, the consumer should be canceled, now with this change, the consumer is re-established, and this is not good because someone could rely on this feature, to have the possibility to remove a queue and stop consuming from them without switch off the entire process that is consuming from that queue.